Panic (nil deref) when creating NewGenericBuffer[any]

As the title states, the library panics with a nil pointer panic when you attempt to create a GenericBuffer with `[any]` as the type.

Minimal example code:
```go
package main

import (
	parquet "github.com/segmentio/parquet-go"
)

func main() {
	parquet.NewGenericBuffer[any]()
}
```

Output:
```
 # ./parquet-panic 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x98 pc=0x6cf1fc]

goroutine 1 [running]:
github.com/segmentio/parquet-go.dereference({0x0?, 0x0?})
	/go/pkg/mod/github.com/segmentio/parquet-go@v0.0.0-20220914210246-f80ef85197ed/schema.go:138 +0x3c
github.com/segmentio/parquet-go.NewGenericBuffer[...]({0x0, 0x722b60?, 0xc0000061a0?})
	/go/pkg/mod/github.com/segmentio/parquet-go@v0.0.0-20220914210246-f80ef85197ed/buffer_go18.go:37 +0x85
main.main()
	/parquet-panic/main.go:8 +0x27
```

According to the docs both `NewGenericBuffer`  _should_ work with `any` as the type parameter. Note that `NewGenericWriter[any]` fails in the same manner.

--- a/buffer_go18.go
+++ b/buffer_go18.go
@@ -33,7 +33,7 @@ func NewGenericBuffer[T any](options ...RowGroupOption) *GenericBuffer[T] {
 	}
 
 	t := typeOf[T]()
-	if config.Schema == nil {
+	if config.Schema == nil && t != nil {
 		config.Schema = schemaOf(dereference(t))
 	}
 
@@ -53,6 +53,9 @@ func typeOf[T any]() reflect.Type {
 type bufferFunc[T any] func(*GenericBuffer[T], []T) (int, error)
 
 func bufferFuncOf[T any](t reflect.Type, schema *Schema) bufferFunc[T] {
+	if t == nil {
+		return (*GenericBuffer[T]).writeRows
+	}
 	switch t.Kind() {
 	case reflect.Interface, reflect.Map:
 		return (*GenericBuffer[T]).writeRows

--- a/buffer_go18_test.go
+++ b/buffer_go18_test.go
@@ -181,3 +181,8 @@ func TestIssue327(t *testing.T) {
 		_ = parquet.NewGenericBuffer[testType]()
 	})
 }
+
+func TestIssue343(t *testing.T) {
+	// must not panic
+	_ = parquet.NewGenericBuffer[any]()
+}

